agent/config/config.go

```diff
@@ -693,3 +694,34 @@ type AuditSink struct {
 	RotateDuration    *string `json:"rotate_duration,omitempty"    hcl:"rotate_duration"    mapstructure:"rotate_duration"`
 	RotateMaxFiles    *int    `json:"rotate_max_files,omitempty"   hcl:"rotate_max_files"   mapstructure:"rotate_max_files"`
 }
+
+type AutoConfigRaw struct {
+	Enabled         *bool                   `json:"enabled,omitempty" hcl:"enabled" mapstructure:"enabled"`
+	IntroToken      *string                 `json:"intro_token,omitempty" hcl:"intro_token" mapstructure:"intro_token"`
+	IntroTokenFile  *string                 `json:"intro_token_file,omitempty" hcl:"intro_token_file" mapstructure:"intro_token_file"`
```

Comment:
I believe the `hcl` and `json` tags in all of these config structs are never used. Only the mapstructure tag is read during config parsing because the hcl/json decodes into a raw `map[string]interface{}`.

We don't necessarily need to fix this now, but while adding new fields might be a good time to do it.

Reply:
I think your right that they are unnecessary. We should remove them all once we know that is the case. Until then, I am just making the new configs look like the old with tags for all.